### DISCUSSION TITLE
How to configure the external dns on the linux crc

### DISCUSSION THREAD

**Discussion by Liubella** [crc-org/crc#3900] [Q&A]
Hi, I am using the crc on linux.  I want to add the external coredns for the pod (192.168.130.22) on the crc.

Step 1: add the  tc_coredns.config file under /etc/NetworkManager/dnsmasq.d
```
[core@crc-2zx29-master-0 dnsmasq.d]$ cat tc_coredns.config
server=/cctf.udm.com/192.168.130.22:5353
```
Step 2: sudo systemctl restart NetworkManager.service
Step 3:  Modify /etc/resolv.conf as following:
```
[core@crc-2zx29-master-0 dnsmasq.d]$ cat /etc/resolv.conf
# Generated by NetworkManager
search crc.testing apps-crc.testing cctf.udm
nameserver 192.168.130.22
nameserver 192.168.130.11
```
Step4:
```
 [core@crc-2zx29-master-0 dnsmasq.d]$ dig  +answer _http._tcp.nrfsrv2.cctf.udm.com SRV

; <<>> DiG 9.16.23-RH <<>> +answer _http._tcp.nrfsrv2.cctf.udm.com SRV
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NXDOMAIN, id: 23724
;; flags: qr rd ra; QUERY: 1, ANSWER: 0, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 4096
;; QUESTION SECTION:
;_http._tcp.nrfsrv2.cctf.udm.com. IN    SRV

;; Query time: 3 msec
;; SERVER: 192.168.130.11#53(192.168.130.11)
;; WHEN: Tue Oct 31 06:06:10 UTC 2023
;; MSG SIZE  rcvd: 60
```
I found the dig still automatically used 192.168.130.11. 

How should I do to solve this issue？

**Comment by praveenkumar** [comment]
We never tried out adding external dns on the crc side, let's hope someone from community can answer this.

**Reply by Liubella** [reply]
Ok, Thanks.
Do we have any way to make node and pod IP (192.168.130.22 or 192.168.126.22) can interconnect?
now ping 192.168.130.22 or 192.168.126.22 from node failed.

**Comment by cfergeau** [comment]
"external coredns" - do you mean this is running in a different VM? Or some place else? I'd  first make sure connectivity from the guest OS to this IP is working fine (firewalld on the host may block this traffic). Then you can look at using this for DNS.

**Reply by Liubella** [reply]
Not with different VM,

1. Start the coredns in one pod. 
2. Config the DNS forwarding in the default dns, and show configmap  as below:

[core@crc-2zx29-master-0 ~]$ oc get configmap/dns-default -n openshift-dns -o yaml
apiVersion: v1
data:
  Corefile: |
    # cctf-dns
    cctf.udm.com:5353 {
        prometheus 127.0.0.1:9153
        forward . 192.168.130.103:5353 {
            policy random
        }
        errors
        log . {
            class error
        }
        bufsize 512
        cache 900 {
            denial 9984 30
        }
    }
    .:5353 {
        bufsize 512
        errors
        log . {
            class error
        }
        health {
            lameduck 20s
        }
        ready
        kubernetes cluster.local in-addr.arpa ip6.arpa {
            pods insecure
            fallthrough in-addr.arpa ip6.arpa
        }
        prometheus 127.0.0.1:9153
        forward . /etc/resolv.conf {
            policy sequential
        }
        cache 900 {
            denial 9984 30
        }
        reload
    }
    hostname.bind:5353 {
        chaos
    }
kind: ConfigMap



But the DNS forwarding it's not worked.
[testrunner@testclient-775c69549-7hvq8 coreDNS]$ nslookup -debug pri.scp.cctf.udm.com.
Server:         10.217.4.10
Address:        10.217.4.10#53

------------
    QUESTIONS:
        pri.scp.cctf.udm.com, type = A, class = IN
    ANSWERS:
    AUTHORITY RECORDS:
    ADDITIONAL RECORDS:
------------
** server can't find pri.scp.cctf.udm.com: SERVFAIL

Any idea for this issue?
